core/src/main/java/io/opentelemetry/android/features/diskbuffering/DiskBufferingConfiguration.java

```diff
@@ -22,6 +22,7 @@ public final class DiskBufferingConfiguration {
     private final long maxFileAgeForWriteMillis;
     private final long minFileAgeForReadMillis;
     private final long maxFileAgeForReadMillis;
+    private final boolean enableDebugMode;
```

Comment:
I think a debug mode should be part of the `OtelRumConfig` or the lowest level of configuration (if there's one) so the whole SDK respects the debug mode flag, and not only for the disk buffering, this would be useful across the SDK and its configurations

Reply:
I think there's a case to be made for both. Top-level debug mode could imply disk buffering debug mode, but not the other way around. In any case, that should be a separate PR IMO. Feel free to log an issue to track.

Reply:
All good, having an SDK debug mode for all options feels too much IMO, a top-level debug mode that logs everything everywhere and respects eg a severity log level for filtering would be cleaner and easier to enable/disable.
If we introduce a debug flag per feature eg diskbuffering, that just adds more complexity everywhere, I'd rather achieve this with TAG filtering if the global debug mode is enabled.

Reply:
I unresolved the comment so that I can copy-paste the comment URL otherwise the anchor link does not work, but it's not that you have to address this in this PR.

Reply:
https://github.com/open-telemetry/opentelemetry-android/issues/668

Reply:
Thanks. I'm not entirely sure I made my point, so let's continue chatting about it at SIG sometime.